### ISSUE TITLE
Cannot compile multiplatform project with both Jetpack Compose and JS with IR compiler

### ISSUE THREAD

**Issue by lucgirardin** [JetBrains/compose-jb#352]
I am trying to create a multiplatform library with both Compose and JS targets. With the JS target in LEGACY mode, everything compiles just fine. However, if I switch to the IR compiler, then the compilation fail with the following error:

```
e: androidx.compose.compiler.plugins.kotlin.IncompatibleComposeRuntimeVersionException: You are using an outdated version of Compose Runtime that is not compatible with the version of the Compose Compiler plugin you have installed. The compose compiler plugin you are using (version 1.0.0-alpha13) expects a minimum runtime version of 1.0.0-alpha13.
        at androidx.compose.compiler.plugins.kotlin.VersionChecker.outdatedRuntimeWithUnknownVersionNumber(VersionChecker.kt:102)
```
While the project is configured to use the latest version of Gradle, Kotlin, and Jetpack Compose, it also fails with earlier versions. I have created a sample project to reproduce the problem:

https://github.com/lucgirardin/mpp-compose-js

Some time ago, I created a thread about this issue in Slack but it did not yield any solution:

https://kotlinlang.slack.com/archives/C01D6HTPATV/p1611065954091900

It would be so cool to be able to mix Compose and JS with expect/actual construct ;-)

Thanks for listening!

Luc

**Comment by jimgoog** [comment]
@olonho Do expect+actual always need to be in the same module?  Or is it possible for an `expect` to live in a different compilation unit from the actual?

@lucgirardin That is to say, I think what is happening here is that you have a single module, where you're trying to use Compose in half the module (desktop) but not in the other half of the module (web).  Is it possible to separate these into different modules and only have `id("org.jetbrains.compose") version "0.3.0-build149"` in the build that is intended for desktop?

If there are technical reasons (eg. expect+actual can't be separated) that they can't be separate modules, then I think we should file a bug on upstream Compose (https://issuetracker.google.com/issues/new?component=612128) stating that the version check should only occur if/when the Compose Compiler actually encounters a composable function, to avoid emitting error messages when the gradle plugin is installed but Compose is not being used.

**Comment by lucgirardin** [comment]
@jimgoog Thanks for looking into this. Alas, having expect and actuals in different module doesn't seem to be possible. During the early days, it used to be possible to use an `expectBy` construct, but that has vanished in recent version of the multiplatform plugin. I could find one open issue asking for this: JetBrains/kotlinconf-app/issues/56.

I have updated my sample project to the latest Compose plugin and have tried other approaches such as applying the Compose plugin only to the JVM section... with no luck.

I will therefore try my luck at pushing the issue upstream, as you suggest.

**Comment by lucgirardin** [comment]
Sent upstream at https://issuetracker.google.com/issues/180633951

**Comment by AlexeyTsvetkov** [comment]
Please note, that the fix will work only with Kotlin 1.5.0-RC+
1.5.0-M1 and 1.5.0-M2 won't work

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6446">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
